earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -264,15 +317,17 @@ void StorageManager<AssetType>::Abort() {
   dirtyMap.clear();
 }
 
-template<class AssetType>
-bool StorageManager<AssetType>::SaveDirtyToDotNew(
+template<class AssetType, class AssetStorageType>
+bool StorageManager<AssetType, AssetStorageType>::SaveDirtyToDotNew(
     khFilesTransaction &savetrans,
     std::vector<AssetKey> *saved) {
   notify(NFY_INFO, "Writing %lu %s records", dirtyMap.size(), assetType.c_str());
   typename std::map<AssetKey, PointerType>::iterator entry = dirtyMap.begin();
   while (entry != dirtyMap.end()) {
     std::string filename = entry->second->XMLFilename() + ".new";
-    if (entry->second->Save(filename)) {
+    AssetSerializerLocalXML<AssetType, AssetStorageType> serializer;
```

Comment:
Minor: might perform slightly better if we declare the serializer outside the loop. Another option is to make the serializer a member of StorageManager so you don't have to keep declaring it.

Reply:
Changed as suggested.